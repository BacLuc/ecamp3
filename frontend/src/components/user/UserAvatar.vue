<template>
  <v-avatar :size="size" :color="color">
    <span class="white--text" :style="style">{{ initials }}</span>
  </v-avatar>
</template>
<script>

export default {
  name: 'UserAvatar',
  props: {
    size: { type: Number, required: false, default: 48 },
    value: { type: Object, required: true }
  },
  computed: {
    user () {
      if (this.value.user instanceof Function) {
        return this.value.user()
      }
      if (this.value.user instanceof Object) {
        return this.value.user
      }
      return this.value
    },
    color () {
      const id = this.user.id
      const h = (parseInt(id, 16) % 360)
      return `hsl(${h}, 100%, 40%)`
    },
    initials () {
      let displayName = ''
      if (this.user.displayName) {
        displayName = this.user.displayName
      } else if (this.value.inviteEmail) {
        displayName = this.value.inviteEmail.split('@', 2).shift()
      }
      let items = displayName.split(' ', 2)
      if (items.length === 1) {
        items = items.shift().split(/[,._-]/, 2)
      }
      if (items.length === 1) {
        return displayName.substr(0, 2)
      } else {
        return items[0].substr(0, 1) + items[1].substr(0, 1)
      }
    },
    style () {
      return {
        fontSize: (this.size / 2.5) + 'px'
      }
    }
  }
}

</script>
