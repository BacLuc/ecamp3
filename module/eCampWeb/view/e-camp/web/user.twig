{% extends 'layout/layout.base.twig' %}

{% block setTitle %}
    {{ parent() }}
    {% do headTitle().append(user.username) %}
{% endblock %}

{% block layout_header %}
    {% include 'header/user.twig' %}
{% endblock %}
