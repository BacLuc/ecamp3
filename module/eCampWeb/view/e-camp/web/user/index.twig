{% extends 'e-camp/web/user.twig' %}

{% block layout_body %}
    <section class="container">

        <div class="row">
            <div class="col-sm-12">
                User...
                {{ dump(user) }}
            </div>
        </div>
    </section>
{% endblock %}
