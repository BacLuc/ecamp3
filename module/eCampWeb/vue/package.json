{
  "name": "ecampweb-vue",
  "version": "1.0.0",
  "description": "JavaScript frontend support using Vue.js for eCampWeb",
  "main": "src/main.js",
  "scripts": {
    "test": "test",
    "dev": "webpack-dev-server --config src/webpack.config.js",
    "build": "webpack --config src/webpack.config.js --env.production"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/ecamp/ecamp3.git"
  },
  "author": "",
  "license": "GPL-3.0-or-later",
  "bugs": {
    "url": "https://github.com/ecamp/ecamp3/issues"
  },
  "homepage": "https://github.com/ecamp/ecamp3#readme",
  "dependencies": {
    "vue": "^2.3.3"
  },
  "devDependencies": {
    "assets-webpack-plugin": "^3.5.1",
    "babel-core": "^6.0.0",
    "babel-loader": "^6.0.0",
    "babel-preset-env": "^1.6.0",
    "css-loader": "^0.25.0",
    "extract-text-webpack-plugin": "^2.1.2",
    "file-loader": "^0.9.0",
    "scss": "^0.2.4",
    "scss-loader": "0.0.1",
    "vue-loader": "^12.1.0",
    "vue-template-compiler": "^2.3.3",
    "webpack": "^2.6.1",
    "webpack-dev-server": "^2.5.0"
  }
}
