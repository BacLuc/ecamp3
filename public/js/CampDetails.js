Vue.component('camp-details', {
    props: ['campId'],
    data: function() {
        return {
            editing: false,
            campDetails: {},
            messages: this.fetchFromAPI(),
        }
    },
    computed: {
        ownerName: function () {
            if (this.campDetails._embedded == null) return '';
            return this.campDetails._embedded.owner.name;
        },
        periods: function () {
            if (this.campDetails._embedded == null) return [];
            return this.campDetails._embedded.periods;
        },
        buttonText: function () {
            return this.editing ? 'Speichern' : 'Bearbeiten';
        },
    },
    methods: {
        fetchFromAPI: function() {
            // Since we don't yet include vue.js with NPM and webpack, we cannot use npm modules such as vue-fetch, and
            // also cannot use advanced javascript features such as async / await (which are required for the
            // fetch API). Therefore, to prove our point, we use axios here to query our eCamp API.
            let $this = this;
            axios.get('/api/camp/' + this.campId)
                .then(function (response) {
                    $this.campDetails = response.data;
                })
                .catch(function (error) {
                    $this.messages = [ { type: 'danger', text: 'Could get camp details. ' + error } ];
                });
        },
        saveToAPI: function() {
            let $this = this;
            axios.put('/api/camp/' + this.campId, this.campDetails)
                .then(function (response) {
                    $this.messages = [ { type: 'success', text: 'Successfully saved' } ];
                    $this.campDetails = response.data;
                })
                .catch(function (error) {
                    $this.messages = [ { type: 'danger', text: 'Could not save camp details' } ];
                });
        },
        toggleEdit: function() {
            if (this.editing) {
                this.saveToAPI();
            }
            this.editing = !this.editing;
        },
    },
    template: '\
    <div>\
        <div v-for="message in messages" role="alert" class="alert" v-bind:class="\'alert-\' + message.type">{{ message.text }}</div>\
        <div class="card" style="margin-bottom: 10px">\
            <div class="card-body">\
                <form v-on:submit.prevent="toggleEdit">\
                    <button type="submit" style="float: right" class="btn btn-sm" v-bind:class="{ \'btn-primary\': editing, \'btn-outline-primary\': !editing }">{{ buttonText }}</button>\
                    Vue.js Infos zu genau einem Lager, id {{ campId }}\
                    <ul>\
                        <li>Name: {{ campDetails.name }}</li>\
                        <li><toggleable-input v-bind:editing="editing" fieldname="Titel" v-model="campDetails.title"></toggleable-input></li>\
                        <li><toggleable-input v-bind:editing="editing" fieldname="Motto" v-model="campDetails.motto"></toggleable-input></li>\
                        <li>Besitzer Name: {{ ownerName }}</li>\
                        <li>Lager-Perioden:\
                            <ul><li v-for="period in periods">{{ period.description }} ({{ period.start }} - {{ period.end }})</li></ul>\
                        </li>\
                    </ul>\
                </form>\
            </div>\
        </div>\
    </div>\
    ',
});

// A component that displays a field as text or as an input field, depending on the editing prop.
// You can two-way bind to the value using v-model.
Vue.component('toggleable-input', {
    props: ['editing', 'fieldname', 'value'],
    computed: {
        valueModel: {
            get: function() {
                return this.value;
            },
            set: function(newValue) {
                this.$emit('input', newValue);
            },
        },
    },
    template: '\
        <span>\
            <span v-bind:style="{display: editing ? \'none\' : \'\'}">{{ fieldname }}: {{ value }}</span>\
            <span v-bind:style="{display: editing ? \'\' : \'none\'}">{{ fieldname }}: <input class="form-control" v-model="valueModel"></span>\
        </span>\
    ',
});

new Vue({ el: '#camp-details' });
