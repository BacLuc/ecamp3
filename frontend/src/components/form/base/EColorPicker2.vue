<!--
Displays a field as a color picker (can be used with v-model)
-->
<template>
  <div
    v-click-outside="{ handler: closePicker, closeConditional: closePickerConditional }"
    class="e-form-container"
  >
    <v-menu
      v-model="pickerOpen"
      transition="scale-transition"
      offset-y
      offset-overflow
      :open-on-click="false"
      :close-on-click="false"
      :close-on-content-click="false"
      min-width="290px"
      max-width="290px"
      @input="onPickerClose"
    >
      <template #activator="{ on }">
        <div v-on="on">
          <EColorField
            ref="input"
            :value="pickerValue"
            :vee-id="veeId"
            :vee-rules="veeRules"
            v-bind="$attrs"
            @input="onInput($event)"
            @click.prevent="onInputClick"
          >
            <template #prepend="{ serializedValue }">
              <ColorSwatch
                ref="inputSwatch"
                :color="serializedValue"
                class="mt-n1"
                :aria-label="
                  pickerOpen ? $tc('global.button.close') : $tc('global.button.open')
                "
                aria-haspopup="true"
                :aria-expanded="pickerOpen ? 'true' : 'false'"
                @click="onInputSwatchClick"
              />
            </template>
          </EColorField>
        </div>
      </template>
      <v-card ref="picker" :ripple="false" tabindex="-1">
        <v-color-picker
          :value="pickerValue"
          :style="{ '--picker-contrast-color': contrast }"
          flat
          @update:color="debouncedPickerValue($event.hex)"
        />
        <v-divider />
        <div class="d-flex gap-2 pa-4 flex-wrap">
          <ColorSwatch
            v-for="swatch in swatches"
            :key="swatch"
            :color="swatch"
            @selectColor="onSwatchSelect($event)"
          />
        </div>
      </v-card>
    </v-menu>
  </div>
</template>

<script>
import { formComponentMixin } from '@/mixins/formComponentMixin.js'
import { contrastColor } from '@/common/helpers/colors.js'
import ColorSwatch from '@/components/form/base/ColorPicker/ColorSwatch.vue'
import { debounce } from 'lodash'

export default {
  name: 'EColorPicker2',
  components: { ColorSwatch },
  mixins: [formComponentMixin],
  inheritAttrs: false,
  props: {
    value: { type: String, required: true },
  },
  emits: ['input'],
  data: () => ({
    pickerOpen: false,
    pickerValue: null,
    debouncedPickerValue: null,
    swatches: [
      '#90B7E4',
      '#6EDBE9',
      '#4dbb52',
      '#FF9800',
      '#FD7A7A',
      '#d584e9',
      '#BBBBBB',

      '#1964B1',
      '#1E86CA',
      '#3DB842',
      '#F1810D',
      '#C71A1A',
      '#CF3BD6',
      '#575757',
    ],
  }),
  computed: {
    contrast() {
      // Vuetify returns invalid value #NANNAN in the initialization phase
      return this.value && this.value !== '#NANNAN' ? contrastColor(this.value) : 'black'
    },
  },
  watch: {
    value: {
      handler(newValue) {
        this.pickerValue = newValue
      },
      immediate: true,
    },
    pickerOpen: {
      handler(open) {
        if (!open) {
          this.onPickerClose()
        }
      },
    },
  },
  created() {
    this.debouncedPickerValue = debounce(this.onInput, 300)
  },
  methods: {
    onInputClick() {
      this.activator = 'input'
      this.pickerOpen = true
    },
    onInputSwatchClick() {
      this.activator = 'swatch'
      this.pickerOpen = !this.pickerOpen
      if (this.pickerOpen) {
        setTimeout(() => {
          this.$refs.picker?.$el.focus()
        }, 100)
      }
    },
    onInput(value) {
      this.pickerValue = value
      this.$emit('input', this.pickerValue)
    },
    onSwatchSelect(color) {
      this.pickerValue = color
      this.$emit('input', this.pickerValue)
      this.pickerOpen = false
      this.$refs.inputSwatch.$el.focus()
    },
    closePicker() {
      this.pickerOpen = false
    },
    closePickerConditional(event) {
      return this.pickerOpen && !this.$refs.picker?.$el.contains(event.target)
    },
    onPickerClose() {
      switch (this.activator) {
        case 'input':
          this.$refs.input.focus()
          break
        case 'swatch':
          this.$refs.inputSwatch.$el.focus()
          break
      }
    },
  },
}
</script>

<style scoped>
:deep(.v-color-picker__dot > div::before) {
  content: 'â€¢';
  color: var(--picker-contrast-color);
  display: block;
  width: 100%;
  line-height: 26px;
  font-size: 28px;
  text-align: center;
}
</style>
