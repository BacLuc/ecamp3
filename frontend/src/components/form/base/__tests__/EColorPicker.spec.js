import { screen, waitFor } from '@testing-library/vue'
import { render, setTestLocale, snapshotOf } from '@/test/renderWithVuetify.js'
import user from '@testing-library/user-event'
import EColorPicker2 from '../EColorPicker2.vue'

import { regex } from 'vee-validate/dist/rules'
import { extend } from 'vee-validate'
extend('regex', regex)

describe('An EColorPicker2', () => {
  const COLOR1 = '#FF0000'
  const COLOR2 = '#FF00FF'
  const COLOR3 = '#FAFFAF'
  const INVALID_COLOR = 'some new color'
  const PICKER_BUTTON_LABEL_TEXT = 'Dialog öffnen, um eine Farbe für test zu wählen'
  const VALIDATION_MESSAGE = 'test is not valid'

  beforeEach(() => {
    setTestLocale('de')
  })

  it('renders the component', async () => {
    // given

    // when
    render(EColorPicker2, {
      props: {
        value: COLOR1,
        label: 'test',
      },
    })

    // then
    await screen.findByDisplayValue(COLOR1)
    screen.getByLabelText(PICKER_BUTTON_LABEL_TEXT)
  })

  it('looks like a color picker', async () => {
    // given

    // when
    const { container } = render(EColorPicker2, {
      props: { value: COLOR1, label: 'test' },
    })

    // then
    expect(snapshotOf(container)).toMatchSnapshot('pickerclosed')

    // when
    await user.click(screen.getByLabelText(PICKER_BUTTON_LABEL_TEXT))

    // then
    await screen.findByTestId('colorpicker')
    expect(snapshotOf(container)).toMatchSnapshot('pickeropen')
  })

  it('opens the picker when the text field is clicked', async () => {
    // given
    render(EColorPicker2, {
      props: { value: COLOR1, label: 'test' },
    })
    const inputField = await screen.findByDisplayValue(COLOR1)

    // when
    await user.click(inputField)

    // then
    await waitFor(async () => {
      expect(await screen.findByTestId('colorpicker')).toBeVisible()
    })
  })

  it('opens the picker when the icon button is clicked', async () => {
    // given
    render(EColorPicker2, {
      props: { value: COLOR1, label: 'test' },
    })
    const button = await screen.getByLabelText(PICKER_BUTTON_LABEL_TEXT)

    // when
    await user.click(button)

    // then
    await waitFor(async () => {
      expect(await screen.findByTestId('colorpicker')).toBeVisible()
    })
  })

  it('closes the picker when clicking outside', async () => {
    // given
    render(EColorPicker2, {
      props: { value: COLOR1, label: 'test' },
    })
    const button = await screen.getByLabelText(PICKER_BUTTON_LABEL_TEXT)
    await user.click(button)
    await waitFor(async () => {
      expect(await screen.findByTestId('colorpicker')).toBeVisible()
    })

    // when
    await user.click(document.body)

    // then
    await waitFor(async () => {
      expect(await screen.queryByTestId('colorpicker')).not.toBeVisible()
    })
  })

  it('closes the picker when pressing escape', async () => {
    // given
    render(EColorPicker2, {
      props: { value: COLOR1, label: 'test' },
    })
    const button = await screen.getByLabelText(PICKER_BUTTON_LABEL_TEXT)
    await user.click(button)
    await waitFor(async () => {
      expect(await screen.findByTestId('colorpicker')).toBeVisible()
    })

    // when
    await user.keyboard('{Escape}')

    // then
    await waitFor(async () => {
      expect(await screen.queryByTestId('colorpicker')).not.toBeVisible()
    })
  })

  it('does not close the picker when selecting a color', async () => {
    // given
    const { container } = render(EColorPicker2, {
      props: { value: COLOR1, label: 'test' },
    })
    const button = await screen.getByLabelText(PICKER_BUTTON_LABEL_TEXT)
    await user.click(button)
    await waitFor(async () => {
      expect(await screen.findByTestId('colorpicker')).toBeVisible()
    })

    // when
    // click inside the color picker canvas to select a different color
    const canvas = container.querySelector('canvas')
    await user.click(canvas, { clientX: 10, clientY: 10 })

    // then
    // close button should stay visible
    await expect(
      waitFor(() => {
        expect(screen.queryByTestId('colorpicker')).not.toBeVisible()
      })
    ).rejects.toThrow(/Received element is visible/)
  })

  it('updates v-model when the value changes', async () => {
    // given
    const { emitted } = render(EColorPicker2, {
      props: { value: COLOR1, label: 'test' },
    })
    const inputField = await screen.findByDisplayValue(COLOR1)

    // when
    await user.clear(inputField)
    await user.keyboard(COLOR2)

    // then
    await waitFor(async () => {
      const events = emitted().input
      // some input events were fired
      expect(events.length).toBeGreaterThan(0)
      // the last one included the parsed version of our entered time
      expect(events[events.length - 1]).toEqual([COLOR2])
    })
    // Our entered time should be visible...
    screen.getByDisplayValue(COLOR2)
    // ...and stay visible
    await expect(
      waitFor(() => {
        expect(screen.getByDisplayValue(COLOR2)).not.toBeVisible()
      })
    ).rejects.toThrow(/Received element is visible/)
  })

  it('updates v-model when a new color is selected in the picker', async () => {
    // given
    const { emitted, container } = render(EColorPicker2, {
      props: { value: COLOR1, label: 'test' },
    })
    await screen.findByDisplayValue(COLOR1)
    const button = await screen.getByLabelText(PICKER_BUTTON_LABEL_TEXT)

    // when
    // click the button to open the picker
    await user.click(button)
    // click inside the color picker canvas to select a different color
    const canvas = container.querySelector('canvas')
    await user.click(canvas, { clientX: 10, clientY: 10 })
    // click the close button
    await user.click(screen.getByTestId('colorpicker'))

    // then
    await waitFor(async () => {
      const events = emitted().input
      // some input events were fired
      expect(events.length).toBeGreaterThan(0)
      // the last one included the parsed version of our entered time
      expect(events[events.length - 1]).toEqual([COLOR3])
    })
    // Our entered time should be visible...
    screen.getByDisplayValue(COLOR3)
    // ...and stay visible
    await expect(
      waitFor(() => {
        expect(screen.getByDisplayValue(COLOR3)).not.toBeVisible()
      })
    ).rejects.toThrow(/Received element is visible/)
  })

  it('validates the input', async () => {
    // given
    render(EColorPicker2, {
      props: { value: COLOR1, path: 'test', validationLabelOverride: 'test' },
    })
    const inputField = await screen.findByDisplayValue(COLOR1)

    // when
    await user.clear(inputField)
    await user.keyboard(INVALID_COLOR)

    // then
    await screen.findByText(VALIDATION_MESSAGE)
  })

  it('accepts 3-digit hex color codes, after picker has been shown', async () => {
    render(EColorPicker2, {
      props: { value: COLOR1, label: 'test' },
    })
    const inputField = await screen.findByDisplayValue(COLOR1)
    const button = await screen.getByLabelText(PICKER_BUTTON_LABEL_TEXT)
    // click the button to open the picker
    await user.click(button)

    // when
    await user.clear(inputField)
    await user.keyboard('#abc')

    // then
    await waitFor(() => {
      screen.getByDisplayValue('#AABBCC')
    })
  })
})
