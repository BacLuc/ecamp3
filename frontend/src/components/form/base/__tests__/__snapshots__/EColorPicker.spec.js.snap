// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`An EColorPicker looks like a color picker: pickerclosed 1`] = `
<div data-app>
     <div class="e-form-container"
          vee-rules="[object Object]">
          <div class="e-form-container"
               value="#ff0000">
               <div class="v-input v-input--hide-details v-input--is-label-active v-input--is-dirty theme--light v-text-field v-text-field--single-line v-text-field--filled v-text-field--is-booted v-text-field--enclosed">
                    <div class="v-input__prepend-outer">
                         <button type="button"
                                 class="v-icon notranslate v-icon--link mdi mdi-palette theme--light"
                                 style="color: rgb(255, 0, 0); caret-color: #ff0000;"></button>
                    </div>
                    <div class="v-input__control">
                         <div class="v-input__slot">
                              <div class="v-text-field__slot">
                                   <input id
                                          type="text">
                              </div>
                         </div>
                    </div>
               </div>
          </div>
          <div class="v-menu">
               <!---->
          </div>
     </div>
</div>
`;

exports[`An EColorPicker looks like a color picker: pickeropen 1`] = `
<div data-app>
     <div class="e-form-container"
          vee-rules="[object Object]">
          <div class="e-form-container"
               value="#ff0000">
               <div class="v-input v-input--hide-details v-input--is-label-active v-input--is-dirty theme--light v-text-field v-text-field--single-line v-text-field--filled v-text-field--is-booted v-text-field--enclosed">
                    <div class="v-input__prepend-outer">
                         <button type="button"
                                 class="v-icon notranslate v-icon--link mdi mdi-palette theme--light"
                                 style="color: rgb(255, 0, 0); caret-color: #ff0000;"></button>
                    </div>
                    <div class="v-input__control">
                         <div class="v-input__slot">
                              <div class="v-text-field__slot">
                                   <input id
                                          type="text">
                              </div>
                         </div>
                    </div>
               </div>
          </div>
          <div class="v-menu">
               <transition-stub name="scale-transition"></transition-stub>
          </div>
     </div>
     <div role="menu"
          class="v-menu__content theme--light menuable__content__active "
          style="max-height: auto; min-width: 290px; max-width: 290px; top: 12px; left: 0px; transform-origin: top left; z-index: 8; display: none;">
          <div class="v-card v-sheet theme--light">
               <div class="v-color-picker v-sheet theme--light v-color-picker--flat theme--light"
                    style="max-width: 300px;">
                    <div class="v-color-picker__canvas"
                         style="width: 300px; height: 150px;">
                         <canvas width="300"
                                 height="150"></canvas>
                         <div class="v-color-picker__canvas-dot"
                              style="width: 10px; height: 10px; transform: translate(295px, -5px);"></div>
                    </div>
                    <div class="v-color-picker__controls">
                         <div class="v-color-picker__preview v-color-picker__preview--hide-alpha">
                              <div class="v-color-picker__dot">
                                   <div style="background: rgb(255, 0, 0);"></div>
                              </div>
                              <div class="v-color-picker__sliders">
                                   <div class="v-input v-color-picker__hue v-input--hide-details theme--light v-input__slider v-color-picker__track">
                                        <div class="v-input__control">
                                             <div class="v-input__slot">
                                                  <div class="v-slider v-slider--horizontal theme--light">
                                                       <input value="0"
                                                              id
                                                              disabled="disabled"
                                                              readonly="readonly"
                                                              tabindex="-1">
                                                       <div class="v-slider__track-container">
                                                            <div class="v-slider__track-background primary lighten-3"
                                                                 style="right: 0px;"></div>
                                                            <div class="v-slider__track-fill primary"
                                                                 style="left: 0px; width: 0%;"></div>
                                                       </div>
                                                       <div role="slider"
                                                            tabindex="0"
                                                            aria-valuemin="0"
                                                            aria-valuemax="360"
                                                            aria-valuenow="0"
                                                            aria-readonly="false"
                                                            aria-orientation="horizontal"
                                                            class="v-slider__thumb-container grey--text text--lighten-2"
                                                            style="left: 0%;">
                                                            <div class="v-slider__thumb grey lighten-2"></div>
                                                       </div>
                                                  </div>
                                             </div>
                                        </div>
                                   </div>
                              </div>
                         </div>
                         <div class="v-color-picker__edit">
                              <div class="v-color-picker__input">
                                   <input type="number"
                                          min="0"
                                          max="255"
                                          step="1">
                                   <span>R</span>
                              </div>
                              <div class="v-color-picker__input">
                                   <input type="number"
                                          min="0"
                                          max="255"
                                          step="1">
                                   <span>G</span>
                              </div>
                              <div class="v-color-picker__input">
                                   <input type="number"
                                          min="0"
                                          max="255"
                                          step="1">
                                   <span>B</span>
                              </div>
                              <button type="button"
                                      class="v-btn v-btn--icon v-btn--round theme--light v-size--small">
                                   <span class="v-btn__content">
                                        <i aria-hidden="true"
                                           class="v-icon notranslate mdi mdi-unfold-more-horizontal theme--light"></i>
                                   </span>
                              </button>
                         </div>
                    </div>
               </div>
               <div class="spacer"></div>
               <button type="button"
                       class="v-btn v-btn--text theme--light v-size--default primary--text">
                    <span class="v-btn__content">
                         Abbrechen
                    </span>
               </button>
               <button type="button"
                       class="v-btn v-btn--text theme--light v-size--default primary--text">
                    <span class="v-btn__content">
                         In Ordnung
                    </span>
               </button>
          </div>
     </div>
</div>
`;
